lib/src/utils/serialization_utils.dart

```diff
@@ -83,4 +178,193 @@ class SerializationUtils {
+      } else {
+        final voteKeyPublicKeyPath = votePublicKeyPath ?? [];
+        assert(votePublicKeyPath != null, 'Missing vote key');
+        assert(VersionCompatibility.checkVersionCompatibility(version).supportsCIP36Vote,
+            'Key derivation path for vote keys not supported by the device');
+        writer.write(serializeDelegationType(CIP36VoteDelegationType.path));
+        writerSerializedPath(writer, voteKeyPublicKeyPath);
+      }
+      return writer.toBytes();
+    });
+  }
+
+  static Uint8List serializeCVoteRegistrationDelegation(ParsedCVoteDelegation delegation) {
+    return useBinaryWriter((ByteDataWriter writer) {
+      writer.write(serializeDelegationType(delegation.type));
+      writer.writeUint32(delegation.weight);
+
+      final void Function() invoker = switch (delegation) {
+        KeyDelegation() => () {
+            writeSerializedHex(writer, delegation.voteKey);
+          },
+        PathDelegation() => () {
+            writerSerializedPath(writer, delegation.voteKeyPath);
+          },
+      };
+
+      invoker();
+      return writer.toBytes();
+    });
+  }
+
+  static Uint8List serializeCVoteRegistrationStakingPath(List<int> stakingPath) {
+    return useBinaryWriter((ByteDataWriter writer) {
+      writerSerializedPath(writer, stakingPath);
+      return writer.toBytes();
+    });
+  }
+
+  static Uint8List serializeCVoteRegistrationPaymentDestination(
+      ParsedOutputDestination paymentDestination, CardanoVersion version) {
+    if (VersionCompatibility.checkVersionCompatibility(version).supportsCIP36) {
+      return serializeTxOutputDestination(paymentDestination, version);
+    } else if (paymentDestination is DeviceOwned) {
+      return serializeAddressParams(paymentDestination.addressParams, version);
+    }
+    throw ValidationException('Invalid payment destination');
+  }
+
+  static Uint8List serializeCVoteRegistrationNonce(String nonce) {
+    return useBinaryWriter((ByteDataWriter writer) {
+      writeSerializedUint64(writer, BigInt.parse(nonce));
+      return writer.toBytes();
+    });
+  }
+
+  static Uint8List serializeCVoteRegistrationVotingPurpose(String? votingPurpose) {
+    return useBinaryWriter((ByteDataWriter writer) {
+      serializeOptionFlag(writer, votingPurpose != null);
+      if (votingPurpose != null) {
+        writeSerializedUint64(writer, BigInt.parse(votingPurpose));
+      }
+      return writer.toBytes();
+    });
+  }
+
+  static Uint8List serializeTxOutputDestination(ParsedOutputDestination destination, CardanoVersion version) {
+    return useBinaryWriter((ByteDataWriter writer) {
+      final typeEncoding = {
+        TxOutputDestinationType.thirdParty: 1,
+        TxOutputDestinationType.deviceOwned: 2,
+      };
+
+      if (destination.type == TxOutputDestinationType.thirdParty ||
+          destination.type == TxOutputDestinationType.deviceOwned) {
+        writer.writeUint8(typeEncoding[destination.type] ?? 0);
+      }
```

Comment:
This will become `writer.writeUint8(destination.type.typeEncoding)` after you add it to the enum. Also, we avoid the `?? 0` because we'll have compile-time guarantee that the typeEncoding value exists.